### ISSUE TITLE
Can get two vis canvases open at the same time

### ISSUE THREAD

**Issue by asalani93** [isenseDev/rSENSE#2201]
**General description:** At the very least, map and photo vis can be on screen at the same time.  This pushes photo vis off the page.

**live/dev/localhost:** live
**iSENSE Version:** 7.2
**Logged in (Y or N):** N
**Admin (Y or N):** N

**OS:** Ubuntu 14.0.4 x64
**Browser/Version:** Chrome 43 x64

**Steps to Reproduce:**
* Create a project.
* Create some datasets.  Make sure each project has a picture associated with it.
* Visualize all of the data.  Make sure that map vis is the default vis.
* Refresh the page and click on Photos as soon as possible.  If you do it right, the divs for the map vis will still be on the page, but there won't be any map.  Photos will also be on screen and pushed to the bottom.

**Associated Image(s):**
![two-canvases](https://cloud.githubusercontent.com/assets/7584534/8334400/befdf712-1a65-11e5-97b5-9160c8656bbc.png)

**Comment by jaypoulz** [comment]
You find the best bugs.

**Comment by pat-son** [comment]
This applies to all vis canvases, not just photos. If you have Map as default and switch vis tabs before the map is rendered, it will push the other canvas to the bottom.
You can also invoke the bug if the Map is not the default. It seems to only happen the first time the map is loaded for that session.

**Comment by jaypoulz** [comment]
The files you'll want to look at are the end functions of the vises (I think baseVis has a hide somewhere in there) and the the runtime code in visualizations.js


My guess is that the bug is in the latter, given the description. Probably some broken / bad logic.

**Comment by pat-son** [comment]
I figured it out. The issue was [this line](https://github.com/isenseDev/rSENSE/blob/26ffc959d69793c622b9b1d244ca73aa4879d9f1/app/assets/javascripts/visualizations/highvis/map.coffee#L355):
`google.maps.event.addListenerOnce(@gmap, 'idle', checkProj)`
If the user switched vis tabs, the listener would still fire, which would cause the map vis to run its update function, and that ended up messing everything up. The solution was simple: just remove that event listener when the user switched tabs. Ended up being a two-line fix.
